### PR TITLE
refactor: move wechat_qrcode to contrib package

### PR DESCRIPTION
**Pull request by dirname**

### CODE REVIEW

**Review by dirname — COMMENTED** [review]
I have tried to move to the contrib package, but in the case of keeping the OpenCV function signature style, some non-exportable functions will be reused.

`toGOStrings`, `Mats_Close`, `CStrings_Close`, `Mats_get`

File: contrib/wechat_qrcode.go [hunk 1]
```diff
@@ -50,3 +52,13 @@ func (wq *WeChatQRCode) DetectAndDecode(img Mat, point *[]Mat) []string {
 
 	return result
 }
+
+func toGoStrings(strs C.CStrings) []string {
```
Comment (dirname) [inline contrib/wechat_qrcode.go:56]: https://github.com/hybridgroup/gocv/pull/952#pullrequestreview-908028319

File: contrib/wechat_qrcode.cpp [hunk 2]
```diff
@@ -14,6 +14,22 @@ StringsVector NewStringsVector() {
...
 }
 
+void WeChatQRCode_CStrings_Close(struct CStrings cstrs) {
+    for ( int i = 0; i < cstrs.length; i++ ) {
+        delete [] cstrs.strs[i];
+    }
+    delete [] cstrs.strs;
+}
+
+void WeChatQRCode_Mats_to(struct Mats mats, int i, Mat dst) {
```
Comment (dirname) [inline contrib/wechat_qrcode.cpp:24]: https://github.com/hybridgroup/gocv/pull/952#pullrequestreview-908028319

File: contrib/wechat_qrcode.cpp [hunk 3]
```diff
@@ -14,6 +14,22 @@ StringsVector NewStringsVector() {
     return new std::vector<std::string>;
 }
 
+void WeChatQRCode_CStrings_Close(struct CStrings cstrs) {
```
Comment (dirname) [inline contrib/wechat_qrcode.cpp:17]: https://github.com/hybridgroup/gocv/pull/952#pullrequestreview-908028319

File: contrib/wechat_qrcode.cpp [hunk 4]
```diff
@@ -14,6 +14,22 @@ StringsVector NewStringsVector() {
...
+        delete [] cstrs.strs[i];
+    }
+    delete [] cstrs.strs;
+}
+
+void WeChatQRCode_Mats_to(struct Mats mats, int i, Mat dst) {
+    mats.mats[i]->copyTo(*dst);;
+}
+
+void WeChatQRCode_Mats_Close(struct Mats mats) {
```
Comment (dirname) [inline contrib/wechat_qrcode.cpp:28]: https://github.com/hybridgroup/gocv/pull/952#pullrequestreview-908028319

**Comment by deadprogram** [comment]
Understood. In a future iteration I want to make it easier to use those from within `contrib`. For now, this is a reasonable solution. Thank you for doing it! Now merging.

**Pull request merged**